draft-ietf-quic-http.md

```diff
@@ -508,113 +508,110 @@ HEADERS frames can only be sent on request / push streams.
-|                [Element Dependency ID (i)]                  ...
+|                     [Placeholder ID (i)]                    ...
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|   Weight (8)  |
-+-+-+-+-+-+-+-+-+
+|  [Weight (8)] | [Priority (8)]|
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 ~~~~~~~~~~
 {: #fig-priority title="PRIORITY frame payload"}
 
 The PRIORITY frame payload has the following fields:
 
-  PT (Prioritized Element Type):
-  : A two-bit field indicating the type of element being prioritized (see
+  P (Prioritized Element Type):
+  : A one-bit field indicating the type of element being prioritized (see
     {{prioritized-element-types}}). When sent on a request stream, this MUST be
-    set to `11`.  When sent on the control stream, this MUST NOT be set to `11`.
+    set to `0`.
+
+  D (Dependent On Placeholder):
+  : A one-bit field indicating whether an element depends upon a placeholder
+    or the root of the tree (see {{element-dependency-types}}).
+
+  W (Weight Indication):
+  : A one-bit field indicating whether an 8-bit weight value is present.
 
-  DT (Element Dependency Type):
-  : A two-bit field indicating the type of element being depended on (see
-    {{element-dependency-types}}).
+  P (Priority Indication):
+  : A one-bit field indicating whether an 8-bit strict priority is present.
 
   Empty:
   : A four-bit field which MUST be zero when sent and has no semantic value on
     receipt.
 
   Prioritized Element ID:
   : A variable-length integer that identifies the element being prioritized.
-    Depending on the value of Prioritized Type, this contains the Stream ID of a
-    request stream, the Push ID of a promised resource, a Placeholder ID of a
-    placeholder, or is absent.
+    The Push ID of a promised resource, a Placeholder ID of a placeholder,
+    or is absent if sent on the request stream.
 
   Element Dependency ID:
-  : A variable-length integer that identifies the element on which a dependency
-    is being expressed. Depending on the value of Dependency Type, this contains
-    the Stream ID of a request stream, the Push ID of a promised resource, the
-    Placeholder ID of a placeholder, or is absent.  For details of
-    dependencies, see {{priority}} and {{!RFC7540}}, Section 5.3.
+  : A variable-length integer that identifies the Placeholder ID on which a
+    dependency is being expressed, or is absent.  For details of dependencies,
+    see {{priority}} and {{!RFC7540}}, Section 5.3.
 
   Weight:
-  : An unsigned 8-bit integer representing a priority weight for the prioritized
-    element (see {{!RFC7540}}, Section 5.3). Add one to the value to obtain a
-    weight between 1 and 256.
+  : An optional unsigned 8-bit integer representing a priority weight for the
+    prioritized element (see {{!RFC7540}}, Section 5.3). Add one to the value
+    to obtain a weight between 1 and 256.  When absent, indicates the resource
+    should be delivered all at once or not at all.
```

Comment:
Server might not have all data in place (e.g., streaming via proxy).  To achieve "delivered all at once or not at all", it seems to me that proxy has to buffer all data, which is impossible if data is unlimited or too large.

Reply:
Maybe a similar indication as in the HTTP/2 RFC makes sense:

"When absent, indicates the resource should be allocated the full available resources from its parent, as long as it is able to make progress."

Reply:
Thanks @rmarx suggestion taken.